### PR TITLE
update engine reference docs with latest changes

### PR DESCRIPTION
**Pull request by thaJeztah**
@usha-mandya ptal

### CODE REVIEW

File: _data/engine-cli/docker_checkpoint.yaml [hunk 1]
```diff
@@ -1,6 +1,98 @@
...
+
+  This feature is currently focused on single-host use cases for checkpoint and
+  restore. Here are a few:
+
+  - Restarting the host machine without stopping/starting containers
+  - Speeding up the start time of slow start applications
+  - "Rewinding" processes to an earlier point in time
+  - "Forensic debugging" of running processes
+
+  Another primary use case of checkpoint & restore outside of Docker is the live
```
Comment (usha-mandya) [inline _data/engine-cli/docker_checkpoint.yaml:32]:
```suggestion
  Another primary use case of checkpoint and restore outside of Docker is the live
```

File: _data/engine-cli/docker_checkpoint.yaml [hunk 2]
```diff
@@ -1,6 +1,98 @@
...
+  abc0123
+
+  $ docker checkpoint create cr checkpoint1
+
+  # <later>
+  $ docker start --checkpoint checkpoint1 cr
+  abc0123
+  ```
+
+  This process just logs an incrementing counter to stdout. If you `docker logs`
```
Comment (usha-mandya) [inline _data/engine-cli/docker_checkpoint.yaml:76]:
```suggestion
  This process just logs an incrementing counter to stdout. If you  run `docker logs`
```

File: _data/engine-cli/docker_checkpoint.yaml [hunk 3]
```diff
@@ -1,6 +1,98 @@
...
+  on disk. Later, the container can be restored from the point it was frozen.
+
+  This is accomplished using a tool called [CRIU](http://criu.org), which is an
+  external dependency of this feature. A good overview of the history of
+  checkpoint and restore in Docker is available in this
+  [Kubernetes blog post](https://kubernetes.io/blog/2015/07/how-did-quake-demo-from-dockercon-work/).
+
+  ### Installing CRIU
+
+  If you use a Debian system, you can add the CRIU PPA and install with apt-get
```
Comment (usha-mandya) [inline _data/engine-cli/docker_checkpoint.yaml:15]:
```suggestion
  If you use a Debian system, you can add the CRIU PPA and install with `apt-get`
```

File: _data/engine-cli/docker_checkpoint.yaml [hunk 4]
```diff
@@ -1,6 +1,98 @@
...
+  ```console
+  Usage:  docker checkpoint create [OPTIONS] CONTAINER CHECKPOINT
+
+  Create a checkpoint from a running container
+
+    --leave-running=false    Leave the container running after checkpoint
+    --checkpoint-dir         Use a custom checkpoint storage directory
+  ```
+
+  And to restore a container:
```
Comment (usha-mandya) [inline _data/engine-cli/docker_checkpoint.yaml:57]:
```suggestion
  To restore a container:
```

Comment (thaJeztah) [reply _data/engine-cli/docker_checkpoint.yaml:58]: Ah, looks like I missed this one; will keep that for a follow-up

**Pull request merged**